src/databricks/labs/ucx/source_code/path_lookup.py

```diff
@@ -0,0 +1,52 @@
+from __future__ import annotations
+
+import os
+import sys
+from collections.abc import Iterable
+from pathlib import Path
+
+
+class PathLookup:
+
+    @classmethod
+    def from_pathlike_string(cls, syspath: str):
+        paths = syspath.split(':')
+        return PathLookup([Path(path) for path in paths])
+
+    @classmethod
+    def from_sys_path(cls):
+        return PathLookup([Path(path) for path in sys.path])
+
+    def __init__(self, paths: list[Path]):
+        self._paths = paths
+        self._cwds = [Path(os.getcwd())]
```

Comment:
```suggestion
    def __init__(self, cwd: Path, sys_path: list[Path]):
        self._sys_path = paths
        self._cwd = cwd
    
    def cd(self, working_directory: Path):
        return PathLookup(working_directory, self._sys_path)
    
    @propertly
    def parent(self):
        return PathLookup(self._cwd.parent, self._sys_path)
```
why do we need multiple cwds?... don't use `os.getcwd()` - it'll lead to undefined behavior, as people won't read the sources to know it's there.

`sys.path` is shared within a notebook context:

![image](https://github.com/databrickslabs/ucx/assets/259697/b5646677-ced7-405f-8ad6-8f8796c66fa3)

Reply:
we don't have multiple cwds, only a stack for push/pop
if we can't use os.getcwd(), how should we populate the initial cwd ?

Reply:
Re `sys.path`, thanks for confirming that it's shared within an entire notebook. But what happens when 'switching' notebook via `%run` or `dbutils.notebook.run` ? Does the target notebook benefit from a fresh `sys.path` or does it inherit the caller's ?